sui_programmability/examples/games/sources/slot_machine.move

```diff
@@ -0,0 +1,91 @@
+///
+module games::slot_machine {
+    use sui::balance::{Self, Balance};
+    use sui::coin::{Self, Coin};
+    use sui::math;
+    use sui::object::{Self, UID};
+    use sui::random::{Self, Random, new_generator};
+    use sui::sui::SUI;
+    use sui::transfer;
+    use sui::tx_context::{Self, TxContext};
+
+    /// Error codes
+    const EInvalidAmount: u64 = 0;
+    const EInvalidSender: u64 = 1;
+    const EInvalidEpoch: u64 = 2;
+
+    /// Game for a specific epoch.
+    struct Game has key, store {
+        id: UID,
+        creator: address,
+        epoch: u64,
+        balance: Balance<SUI>,
+    }
+
+    /// Create a new game with a given initial reward for the current epoch.
+    public fun create(
+        reward: Coin<SUI>,
+        ctx: &mut TxContext
+    ) {
+        let amount = coin::value(&reward);
+        assert!(amount > 0, EInvalidAmount);
+        transfer::public_share_object(Game {
+            id: object::new(ctx),
+            creator: tx_context::sender(ctx),
+            epoch: tx_context::epoch(ctx),
+            balance: coin::into_balance(reward),
+        });
+    }
+
+    /// Creator can withdraw remaining balance if the game is over.
+    public fun close(game: &mut Game, ctx: &mut TxContext): Coin<SUI> {
+        assert!(tx_context::epoch(ctx) > game.epoch, EInvalidEpoch);
+        assert!(tx_context::sender(ctx) == game.creator, EInvalidSender);
+        let full_balance = balance::value(&game.balance);
+        coin::take(&mut game.balance, full_balance, ctx)
+    }
+
+    /// Play one turn of the game.
+    ///
+    /// The function consumes the same amount of gas independently of the random outcome.
+    entry fun play(game: &mut Game, r: &Random, coin: &mut Coin<SUI>, ctx: &mut TxContext) {
+        assert!(tx_context::epoch(ctx) == game.epoch, EInvalidEpoch);
+        assert!(coin::value(coin) > 0, EInvalidAmount);
+
+        // play the game
+        let generator = new_generator(r, ctx);
+        let bet = random::generate_u8_in_range(&mut generator, 1, 100);
+        let won = 1 - bet / 50; // equals 1 with probability 49% and 0 otherwise
+
+        // move the bet amount from the user's coin to the game's balance
+        let coin_value = coin::value(coin);
+        let bet_amount = math::min(coin_value, balance::value(&game.balance));
+        coin::put(&mut game.balance, coin::split(coin, bet_amount, ctx));
+
+        // move the reward to the user's coin
+        let reward = 2 * (won as u64) * bet_amount;
+        // the assumption here is that the next line does not consumes more gas when called with zero reward than with
+        // non-zero reward
+        coin::join(coin, coin::take(&mut game.balance, reward, ctx));
+    }
+
+    #[test_only]
+    public fun get_balance(game: &Game): u64 {
+        balance::value(&game.balance)
+    }
+
+    #[test_only]
+    public fun get_epoch(game: &Game): u64 {
+        game.epoch
+    }
+
+}
```

Comment:
Other example ideas:
- NFT with a random element inside
- One round of children's card game "war"--two players agree to play + put up $$, deck gets shuffled, a card drawn for each player, winner gets the funds

Reply:
Added an NFT that can be airdropped without a value and later converted to a random NFT by users.